File: gcd/gcd.cpp
#include <iostream>

using namespace std;

int gcd_bits(int a, int b)
{
    while(b){
        b^=a^=b^=a%=b;
    }
    return a;
}

int gcd_loop(int a, int b)
{
    int temp;

    while(b != 0)
    {
        temp = b;
        b = a%b;
        a = temp;
    }

    return a;
}

int gcd_recursive(int a, int b)
{
    if (b == 0) {
        return a;
    }
    return gcd_recursive(b, a%b);
}


int main()
{
    cout << gcd_bits(66, 24) << endl;
    cout << gcd_loop(66, 24) << endl;
    cout << gcd_recursive(66, 24) << endl;

    return 0;
}

File: gcd/gcd.go
package main

import (
	"fmt"
)

func gcd_bits(a, b int) int {
	for b != 0 {
		a %= b
		b ^= a
		a ^= b
		b ^= a
	}
	return a
}

func gcd_loop(a, b int) int {
	for b != 0 {
		a, b = b, a%b
	}
	return a
}

func gcd_recursive(a, b int) int {
	if b == 0 {
		return a
	}
	return gcd_recursive(b, a%b)
}

func main() {
	fmt.Println(gcd_bits(66, 24))
	fmt.Println(gcd_loop(66, 24))
	fmt.Println(gcd_recursive(66, 24))
}
